lib/apartment/adapters/mysql2_adapter.rb

```diff
@@ -32,6 +32,19 @@ def current_database_name=(name)
         Thread.current[:apartment_current_database_name] = name
       end
 
+      # Override process method to ensure current database is restored propertly
+      def process(database_config = nil, use_use=true)
+        saved_config = current_database.merge({target_database: current_database_name})
```

Comment:
current_database is delegated to ActiveRecord and, thus, at a minimum pulls the default db config specified in database.yml. There should be no case that we call this an no config exists.